synapse/assets/storm/migrations/model-0.2.31.storm

```diff
@@ -0,0 +1,511 @@
+$layers = $lib.spooled.set()
+$layers.adds($layridens)
+
+$views = $lib.view.list(deporder=(true))
+$rviews = $lib.view.list(deporder=(true))
+$rviews.reverse()
+
+$refinfoCache = ({})
+
+function getRefInfo(formname) {
+    $cached = $refinfoCache.$formname
+    if $cached { return($cached) }
+
+    $refinfo = ([])
+
+    syn:prop:type*in=($formname, array, ndef)
+    // Don't include the form we're querying
+    -:form = $formname
+
+    $prop = $lib.model.prop($node.repr())
+    $proptype = $prop.type.name
+
+    // Don't include array types that don't have a base type of $formname or ndef
+    -$($proptype = 'array' and not ([$formname, 'ndef']).has($prop.type.opts.type))
+
+    $isarray = $lib.false
+
+    if ($proptype = 'array') {
+        $isarray = $lib.true
+        $proptype = $prop.type.opts.type
+    }
+
+    $refinfo.append((:form, $prop.name, $proptype, $isarray, :ro))
+
+    fini {
+        $refinfoCache.$formname = $refinfo
+        return($refinfo)
+    }
+}
+
+function getRoProps(form) {
+    $ret = ([])
+
+    syn:prop:form=$form
+    +:ro=(true)
+    $ret.append(:base)
+
+    fini { return($ret) }
+}
+
+function _storeNode(n, viewiden, layriden) {
+
+    $nodeq = $lib.queue.gen("model_0_2_31:nodes")
+    $refsq = $lib.queue.gen("model_0_2_31:nodes:refs")
+    $edgeq = $lib.queue.gen("model_0_2_31:nodes:edges")
+    $editq = $lib.queue.gen("model_0_2_31:nodes:edits")
+
+    $iden = $n.iden()
+    $repr = $n.repr()
+    $form = $n.form()
+    $ndef = $n.ndef()
+
+    $offsets = ({
+        "refs": ([]),
+        "edges": ([]),
+        "edits": ([]),
+    })
+
+    $lib.log.debug(`Saving invalid {$form} node info: {$repr}`)
```

Comment:
For the logs that are made per-node, lets add the node iden to them when possible. The following extra data ends up in structured log output:
```
`, extra=({'iden': $niden'})
```